### PR TITLE
Swallow python warnings during ast.parse (fix #606)

### LINKED ISSUE #606: Swallow python warnings when running ast.parse
`check_duplicate_functions` uses `ast.parse` which, sometimes write python warnings (ex. DeprecationWarnings). 
Those warnings missing some important information, like original filename.

Slash should swallow those warnings.

### PR DESCRIPTION
**Pull request by ayalash**

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Changelog
 =========
 
+* :bug:`606` Swallow python warnings during ast.parse
```
Comment (vmalloc) [inline doc/changelog.rst]: @ayalash should be `:bug major:` so that it appears in the changelog properly

**Comment by ayalash** [comment]
@vmalloc I pushed a fix for your comment. Please review again.

**Comment by vmalloc** [comment]
@ayalash still not seeing it... Are you sure?

**Comment by ayalash** [comment]
Is this is the commit you seeing?
https://github.com/getslash/slash/commit/204fffdfd76129d8c2d167a7929006689630f4b5

**Comment by vmalloc** [comment]
@ayalash now it is... Weird - it was the old one up to until ~5 minutes ago

**Comment by ayalash** [comment]
@vmalloc You approved the changes but didn't merge this PR. Should I change anything else?

**Comment by vmalloc** [comment]
@ayalash the Travis build is still ongoing....

**Comment by vmalloc** [comment]
But I'll merge it anyway because it seems to take forever

**Comment by ayalash** [comment]
@vmalloc Sorry - I didn't noticed it is still running (I got email that it finished successfully, but it probably was the previous commit).

Thanks :)

**Pull request merged**